### DISCUSSION TITLE
Debug (Not available) on Performance Analyzer query

### DISCUSSION THREAD

**Discussion by Mike-Honey** [TabularEditor/TabularEditor3#853] [🙋♀️TE3 Q&A]
1st time using this tool so possible newbie error. 
I connected to my Power BI Desktop session using Model From DB / Local Instance. Not sure I followed the right steps as the terminology was confusing.
In PBI Desktop I used the Performance Analyzer in Power BI to capture a slow query for a matrix visual and pasted it into TE, hit execute and it ran. A bit stumped then, but eventually realised there were 2 results windows, and the data for my matrix was on the 2nd one.

I right-clicked in the results pane, but for every cell it displays 'Debug (not available)'. The measures used are all explicit.

I'm stuck.

**Comment by edhans** [comment]
Yes, DAX queries can generate multiple results, and the one you are _usually_ interested in is the last. I've seen a visual generate 3 results to get the data it needs to render.

I've also run into issues with Debug not being available, and it isn't always clear to me why that happens. Try to remove all measures from the visual except the one you are interested in and see if debug becomes available. 

Just curious, do you have any Calculation Groups acting on this visual?

**Comment by mlonsk** [comment]
Hi @Mike-Honey 
Glad to hear you are getting into Tabular Editor.
With regard to the lack of debugging options, this is most likely due to some of the limitations that we currently have with the debugger: [DAX Debugger Limitations](https://docs.tabulareditor.com/te3/features/dax-debugger.html#limitations-and-known-issues)
These are issues that we plan to address in the future, but I currently do not have a timeline. 

Does your DAX query from the performance analyser contain any of those known issues?

When I use the performance analyser queries, I prefer to strip as much away as possible to only have the core query left and then debug that.

**Comment by sofdit76139** [comment]
I am having the same issue, and I'm not sure why it's occurring. I did look at the limitations, but I can't figure out why debugging doesn't work. This is the performance analyzer query:

```
// DAX Query
DEFINE
	VAR __DS0FilterTable = 
		FILTER(
			KEEPFILTERS(VALUES('02 Dates'[Date])),
			AND('02 Dates'[Date] >= DATE(2021, 10, 7), '02 Dates'[Date] < DATE(2023, 10, 7))
		)

	VAR __DS0FilterTable2 = 
		FILTER(
			KEEPFILTERS(VALUES('Asset'[Reporting Name])),
			NOT('Asset'[Reporting Name] IN {BLANK()})
		)

	VAR __DS0FilterTable3 = 
		TREATAS({TRUE}, 'Asset'[Active Asset])

	VAR __ValueFilterDM2 = 
		FILTER(
			KEEPFILTERS(
				SUMMARIZECOLUMNS(
					'Asset'[Reporting Name],
					'Asset'[SerialNumber],
					__DS0FilterTable,
					__DS0FilterTable2,
					__DS0FilterTable3,
					"Underutilized_Asset_Count", IGNORE('01 All Measures'[Underutilized Asset Count]),
					"v__of_Fleet_Underutilized", '01 All Measures'[% of Fleet Underutilized]
				)
			),
			[v__of_Fleet_Underutilized] > 0
		)

	VAR __DS0Core = 
		SUMMARIZECOLUMNS(
			'Asset'[Reporting Name],
			__DS0FilterTable,
			__DS0FilterTable2,
			__DS0FilterTable3,
			__ValueFilterDM2,
			"v__of_Fleet_Underutilized", '01 All Measures'[% of Fleet Underutilized],
			"Underutilized_Asset_Count", IGNORE('01 All Measures'[Underutilized Asset Count])
		)

	VAR __DS0PrimaryWindowed = 
		TOPN(1001, __DS0Core, [v__of_Fleet_Underutilized], 0, 'Asset'[Reporting Name], 1)

EVALUATE
	__DS0PrimaryWindowed

ORDER BY
	[v__of_Fleet_Underutilized] DESC, 'Asset'[Reporting Name]
```

**Reply by otykier** [reply]
@sofdit76139 I tried to recreate this, but was not able to reproduce the error. Are you using Tabular Editor v. **3.11.0**? Also, what error message are you getting?
